src/main/java/org/shanerx/backup/AutoBackup.java

```diff
@@ -151,41 +160,77 @@ public void run() {
-                            }, mode.getCompressionLevel());
-                    }
-                    else {
+                        }, mode.getCompressionLevel());
+                    } else {
                         ZipUtil.pack(mode.getDir(), zipFile, s -> {
 
                             if (!rec && s.split("/").length > 1) {
                                 return null;
                             }
                             return s;
 
-                            },  mode.getCompressionLevel());
+                        }, mode.getCompressionLevel());
+                    }
+
+                    if (log) {
+                        getServer().getConsoleSender().sendMessage(Message.BACKUP_SUCCESSFUL.toConsoleString()
+                                + mode.getName());
                     }
 
-                    if (log) getServer().getConsoleSender().sendMessage(Message.BACKUP_SUCCESSFUL.toConsoleString()
-                            + mode.getName());
-                } catch(Exception e){
+                    autoPurgeBackups(mode);
+
+                } catch (Exception e) {
                     e.printStackTrace();
-                    if (log) getServer().getConsoleSender().sendMessage(Message.BACKUP_FAILED.toConsoleString()
-                            + mode.getName());
+                    if (log)
+                        getServer().getConsoleSender().sendMessage(Message.BACKUP_FAILED.toConsoleString()
+                                + mode.getName());
                     success[0] = false;
                     failReason[0] = e.getMessage();
                 }
             }
         };
 
-        if (async) runnable.runTaskAsynchronously(this);
-        else runnable.runTask(this);
+        if (async)
+            runnable.runTaskAsynchronously(this);
+        else
+            runnable.runTask(this);
 
         logToFile(success[0] ? BackupAction.SUCCESS : BackupAction.FAIL, failReason[0], logEntity, zipName);
         return success[0];
     }
 
-    public boolean purgeBackups(String logEntity) {
+    private void autoPurgeBackups(BackupMode mode) throws IOException {
+        if (mode.getPurgeAfter() <= 0) {
+            return;
+        }
+
+        Set<BackupFile> backups = getPastBackups(backupsDir.getAbsolutePath());
+        Set<BackupFile> backupsForThisMode = backups.stream()
+                .filter(currentBackup -> currentBackup.getMode().equals(mode.getName()))
+                .collect(Collectors.toSet());
+
+        for (BackupFile backup : backupsForThisMode) {
+            LocalDateTime backupDate = backup.getBackupDate();
+
+            long backupTimeMilliSeconds = backupDate.toInstant(ZoneOffset.ofTotalSeconds(0)).toEpochMilli();
+            long millisecondsNow = LocalDateTime.now().toInstant(ZoneOffset.ofTotalSeconds(0)).toEpochMilli();
+            long purgeTime = backupTimeMilliSeconds + mode.getPurgeAfter() * 1000 * 60;
+            if (millisecondsNow < purgeTime) {
```

Comment:
If purgeTime is 0 (disabled), the backup will be deleted.

Reply:
Hm, couldve sworn that this last fix was already in. Will do! :D

Reply:
Nope, it's not! Have a look at line 203, there, I pre-check for auto-purging. if its smaller than or equal to zero, all autopurging is skipped :)

Reply:
Ah! Missed that, thanks